# Week 1 Genome Assembly -- Feedback

0.5 + 0.5 + 0.5 + 0 + 0.25 + 0.75 + 0.5 + 0.5 + 0.5 + 0.5 = 4.5 points out of 10 possible points

1. Question 1.1, 1.4 how many reads (0.5 pts each)

  * Only see answer for 5x --> +0.5

2. Question 1.2, 1.4 simulation script(s)

  * Only see simulation for 5x --> +0.5

3. Question 1.2, 1.4 plotting script(s)

  * See plotting for both, but you are plotting the simulation results from 5x twice --> +0.5

4. Question 1.2, 1.4 histograms with overlaid Poisson distributions (0.5 pts each)

  * please submit your plots

5. Question 1.3, 1.4 how much of genome not sequenced/comparison to Poisson expectations (0.5 pts each, 0.25 for answer and 0.25 for code)

  * See observation for 5x only, what's the Poisson expectation? and 15X? --> +0.25

6. Question 2 De novo assembly (0.5 pts each, 0.25 for answer and 0.25 for code)

  * number of contigs --> +0.25, code?
  * total length of contigs --> +0.5

7. Question 2 De novo assembly cont (0.5 pts each, 0.25 for answer and 0.25 for code)

  * size of largest contig --> +0.5, code?
  * contig n50 size --> you want to order the contigs by length, largest to smallest, and then sum to find the contig whose length is shorter than or equal to other contigs larger than it, such that the sum of these contigs is half the genome length.

8. whole genome alignment (0.33 pts each, 0.33/2 for answer and 0.33/2 for code)

  * average identity --> +0.33/2, code?
  * length of longest alignment --> +0.33/2, code?
  * how many insertions and deletions in assembly --> +0.33/2, code?

9. decoding the insertion (0.5 pts each, 0.25 for answer and 0.25 for code)

  * position of insertion in assembly --> this doesn't match with what I would expect, and how did you find it?
  * length of novel insertion --> 0.5

10. decoding the insertion cont (0.5 pts each, 0.25 for answer and 0.25 for code)

  * DNA sequence of encoded message --> +0.25, code is the right idea
  * secret message --> +0.25, code is the right idea

Your commands and given output, as you have them written down, would not produce the correct output message. We're okay with you working with other people, but we want to make sure that you also understand the commands that you're running. If your code isn't working, you're always welcome to reach out for help, either from classmates or from the TAs. So, if you really did get the correct output, please correct the commands you used, but if you were having trouble getting that output, let us know and we can help.
